ws-conn/ws-connection-handler.go

```diff
@@ -93,9 +103,13 @@ func (handler *ConnectionHandlerImpl) readDataFromConnections(inputChan chan []b
 	defer handler.removeConnection(wsConn)
 
 	for {
+		wsConn.SetPongHandler(func(string) error { wsConn.SetReadDeadline(time.Now().Add(pongWait)); return nil })
 		msgType, wsBytes, err := wsConn.ReadMessage()
 		if err != nil {
-			logrus.Errorf("failed to read ws-conn message. Cause: %v", err)
+			if IsNormalWSError(err) {
+				logrus.Errorf("failed to read ws-conn message. Cause: %v", err)
+			}
+			handler.removeConnection(wsConn)
```

Comment:
There is already `defer handler.removeConnection(wsConn)` at the beginning of the method. Doesn't it mean that it will be called twice?